Aesop/Index/Basic.lean

```diff
@@ -11,14 +11,14 @@ open Lean.Meta
 
 namespace Aesop
 
--- This value controls whether we use 'powerful' reductions, e.g. iota, when
--- indexing Aesop rules. See the `DiscrTree` docs for details.
-def simpleReduce := false
+-- This value controls which reductions are used when normalising expressions
+-- stored in the discrimination trees. We use the same config as `simp`.
+def discrTreeConfig : WhnfCoreConfig := { iota := false, proj := .no }
```

Comment:
This is a behaviour change:  the previous `simpleReduce := false` corresponds to `{}`. The current config is the one `simp` uses, which seems like it might be correct, but I don't really understand the implications of these options. I'm curious to see whether mathlib breaks due to this change.

Reply:
Mathlib seems to be okay with it! This is probably going to hit `nightly-testing` today ...

Reply:
Okay, if it doesn't break mathlib then I'm happy with the change.